src/libopensc/pkcs15-sec.c

```diff
@@ -287,7 +287,15 @@ int sc_pkcs15_derive(struct sc_pkcs15_card *p15card,
 
 	/* If card stores derived key on card, then no data is returned
 	 * and the key must be used on the card. */
-	*poutlen = r;
+	
+	switch (obj->type) {
+		case SC_PKCS15_TYPE_PRKEY_EC:
+	/* But in case the card returned the shared secret, don't overwrite *poutlen */
+			break;
+		default:
+			*poutlen = r;
+	}
+	
```

Comment:
Thanks for your comment. This code was added due to a misunderstanding. Actually the problem was in the myeid driver where functions myeid_decipher and myeid_ecdh_derive returned SC_SUCCESS instead of the data length, so we got 0 from use_key and thought that the problem is here. I will remove this code and fix the myeid driver.